server/apps/archive_broadcast/broadcast.py

```diff
@@ -293,5 +295,71 @@ def spike_item(self, original):
         If Original item is re-write then it will remove the reference from the broadcast item.
         :param: dict original: original document
         """
-        if original.get('rewrite_of'):
+        broadcast_items = [item for item in self.get_broadcast_items_from_master_story(original)
+                           if item.get(ITEM_STATE) not in PUBLISH_STATES]
+
+        if broadcast_items:
+            for item in broadcast_items:
+                id_ = item.get(config.ID_FIELD)
+                try:
+                    self.packageService.remove_spiked_refs_from_package(id_)
+                    updates = {ITEM_STATE: CONTENT_STATE.SPIKED}
+                    resolve_document_version(updates, SOURCE, 'PATCH', item)
+                    get_resource_service('archive_spike').patch(id_, updates)
+                    insert_into_versions(id_=id_)
+                except:
+                    logger.exception(message="Failed to spike the related broadcast item {}.".format(id_))
+
+        if original.get('rewrite_of') and original.get(ITEM_STATE) not in PUBLISH_STATES:
             self.remove_rewrite_refs(original)
+
+    def kill_broadcast(self, updates, original):
+        """
+        "Kill the broadcast items
+        :param dict updates:
+        :param dict original:
+        :return:
+        """
+        broadcast_items = [item for item in self.get_broadcast_items_from_master_story(original)
+                           if item.get(ITEM_STATE) in PUBLISH_STATES]
+
+        for item in broadcast_items:
+            item_id = item.get(config.ID_FIELD)
+            packages = self.packageService.get_packages(item_id)
+
+            if packages.count() > 0:
+                processed_packages = []
```

Comment:
If the expected number of packages is on the high side, it might make sense to use a `set()` for this lookup collection.